File: interfaces/matlab/hpipm_matlab/+hpipm_matlab/hpipm_solver.m
function solver = hpipm_solver(m_qp_dims, m_qp_data)
    import hpipm_matlab.*
    hp = py.importlib.import_module('hpipm_python');
    py_qp_dims = hp.hpipm_dims();
    py_qp_dims = m2py_obj(m_qp_dims, py_qp_dims);
    
    py_qp_data = hp.hpipm_data();
    py_qp_data = m2py_obj(m_qp_data, py_qp_data);
    solver = hp.hpipm_solver(py_qp_dims, py_qp_data);
    
end

File: interfaces/julia/hpipm_solver.jl
function hpipm_data()
    py_qp_data = hp.hpipm_data()
    ju_qp_data = py2ju_obj(py_qp_data)
    return ju_qp_data
end

function hpipm_dims()
    py_qp_dims = hp.hpipm_dims()
    ju_qp_dims = py2ju_obj(py_qp_dims)
    return ju_qp_dims
end
